synapse/config/server.py

```diff
@@ -35,6 +35,14 @@
 
 logger = logging.Logger(__name__)
 
+# an object to pass to the "default" parameter, with which we can
+# `is` against to be sure we have an undefined config option.
+UNDEFINED = object()
```

Comment:
Is `UNDEFINED` purely internal to this file?
ie. Will outside code never observe `ServerConfig.worker_to_update_user_directory` with this value?

Reply:
No, though i'll add an `assert` to make that sure